src/lib/users/clients/inst_user_first.rb

```diff
@@ -0,0 +1,699 @@
+                Popup.YesNoHeadline(
+                  # yes-no popup headline
+                  _("Empty User Login"),
+                  # yes-no popup contents
+                  _("Leaving the user name empty only makes sense\n" +
+                    "in a network environment with an authentication server.\n" +
+                    "Leave it empty?")
+                )
+              break
+            else
+              next
+            end
+          end
+
+          if !valid_username?(@username)
+            UI.SetFocus(Id(:username))
+            next
+          end
+
+          # full name checks
+          @full_name = UI.QueryWidget(Id(:full_name), :Value)
+          @error = UsersSimple.CheckFullname(@full_name)
+          if !@error.empty?
+            Report.Error(@error)
+            UI.SetFocus(Id(:full_name))
+            next
+          end
+
+          # password checks
+          pw1 = UI.QueryWidget(Id(:pw1), :Value)
+          pw2 = UI.QueryWidget(Id(:pw2), :Value)
+          @use_pw_for_root = UI.QueryWidget(Id(:root_pw), :Value)
+
+          if !valid_password?(@username, pw1, pw2)
+            UI.SetFocus(Id(:pw1))
+            next
+          end
+
+          # set UID if home directory is found on future home partition
+          @password = pw1
+        end
+
+        if @ret == :abort && !Popup.ConfirmAbort(:painless)
+          next
+        end
+
+        break if [:back, :abort, :cancel, :next].include?(@ret)
+      end
+
+      if @ret == :next
+        UsersSimple.SetAfterAuth("users")
+        UsersSimple.SetKerberosConfiguration(false)
+        if !@to_import.empty?
+          create_users = []
+
+          @to_import.each do |name|
+            u = @imported_users.fetch(name, {})
+            u["__imported"] = true
+            u["encrypted"] = true
+            create_users << u
+          end
+
+          UsersSimple.SetUsers(create_users)
+          UsersSimple.SkipRootPasswordDialog(false)
+          UsersSimple.SetRootPassword("") if @root_dialog_follows
+          UsersSimple.SetAutologinUser("")
+          UsersSimple.SetRootAlias("")
+        elsif !@username.empty?
+          # save the first user data
+          @user_map = {
+            "uid"          => @username,
+            "userPassword" => @password,
+            "cn"           => @full_name
+          }
+          UsersSimple.SetUsers([@user_map])
+          UsersSimple.SkipRootPasswordDialog(@use_pw_for_root)
+          if @root_dialog_follows || @use_pw_for_root
+            UsersSimple.SetRootPassword(@use_pw_for_root ? @password : "")
+          end
+
+          UsersSimple.SetAutologinUser(
+            UI.QueryWidget(Id(:autologin), :Value) == true ? @username : ""
+          )
+          UsersSimple.SetRootAlias(
+            UI.QueryWidget(Id(:root_mail), :Value) == true ? @username : ""
+          )
+        end
+      end
+
+      Wizard.CloseDialog if Mode.normal
+      Progress.set(@progress_orig)
+      @ret
+    end
```

Comment:
This method hurts my eyes and heart, can it be work accident?

At least please separate some method to lazy loaded attributes in own method.